design/slot-reservations.md

```diff
@@ -0,0 +1,225 @@
+can reserve the slot. Only hosts that have reserved the slot can fill the slot.
+After the host downloads the content and calculates a proof, it can move the
+slot from its reserved state into the filled state by providing collateral and
+the storage proof. Then it begins to periodically provide storage proofs and
+accrue payments for the slot.
+
+```
+         reserve         proof & collateral
+            |                  |
+            v                  v
+            ---------------------------------------------
+     slot:  |/ / / / / / / / / |/////////////////////////
+            ---------------------------------------------
+            |                  |
+            v                  v
+          slot                slot
+         reserved            filled
+
+
+            ---------------- time ---------------->
+```
+
+There is an initial race for eligible SPs who are first to secure a reservation, then a
+second race amongst the SPs with a reservation to fill the slot (with collateral
+and the generated proof). However, not all SPs in the network can reserve a slot
+initially: the [expanding window
+mechanism](https://github.com/status-im/codex-research/blob/ad41558900ff8be91811aa5de355148d8d78404f/design/marketplace.md#dispersal)
+dictates which SPs are eligible to reserve the slot.
+
+### Expanding window mechanism
+
+The expanding window mechanism prevents node and network overload once a slot
+becomes available to be filled (or repaired) by allowing only a very small
+number of SP addresses to fill/repair the slot at the start. Over time, the
+number of eligible SP addresses increases, until eventually all SP addresses in
+the network are eligible.
+
+The expanding window mechanismstarts off with a random source address, defined
+as $hash(block hash, request id, slot index, reservationindex)$. The distance
+between each SP address and the source address can be defined as $XOR(A, A_0)$
+(kademlia distance). Once the allowed distance is greater than SP's distance,
+the SP is considered eligible to reserve a slot. The allowed distance for
+eligible addresses over time $t_i$ can be [defined as $2^{256} *
+F(t_i)$](https://hackmd.io/@bkomuves/BkDXRJ-fC). As this allowed distance value
+increases along a curve, more and more addresses will be eligible to participate
+in reserving that slot. In total, eligible addresses are those that satisfy:
+
+$XOR(A, A_0) < 2^{256} * F(t_i)$
+
+Because the source address for the expanding window is generated using the slot
+index and the reservation index, that means the source address for each
+reservation in each slot will be different.
+
+Furthermore, the client can set the rate of expansion by defining the [parameter
+$h$](https://hackmd.io/@bkomuves/BkDXRJ-fC#Parametrizing-the-speed-of-expansion)
+as the [dispersal parameter](https://github.com/codex-storage/codex-research/blob/ad41558900ff8be91811aa5de355148d8d78404f/design/marketplace.md#dispersal) of the request.
```

Comment:
```suggestion
Furthermore, the client can set the rate of expansion by defining the parameter
$h$ ([more information](https://hackmd.io/@bkomuves/BkDXRJ-fC#Parametrizing-the-speed-of-expansion))
as the [dispersal parameter](https://github.com/codex-storage/codex-research/blob/ad41558900ff8be91811aa5de355148d8d78404f/design/marketplace.md#dispersal) of the request.
```

Reply:
I updated this section a bit, and removed any links on equations.